File: src/types/agent.ts
// Types pour le module Agent/User

/**
 * Région de Côte d'Ivoire
 */
export interface Region {
  id: string;
  code: string;
  nom: string;
  description?: string;
  createdAt: Date;
  updatedAt: Date;
}

/**
 * Agent de collecte
 */
export interface Agent {
  id: string;
  code: string;
  nom: string;
  prenom: string;
  email?: string;
  telephone: string;
  statut: 'actif' | 'inactif' | 'suspendu';
  
  // Identité
  date_naissance?: Date;
  lieu_naissance?: string;
  nationalite?: string;
  type_piece?: string;
  numero_piece?: string;
  photo?: string;
  
  // Relations
  regions?: AgentRegion[];
  operations?: any[];
  
  createdAt: Date;
  updatedAt: Date;
}

/**
 * Affectation Agent-Région
 */
export interface AgentRegion {
  id: string;
  id_agent: string;
  id_region: string;
  agent?: Agent;
  region?: Region;
  date_affectation: Date;
  date_fin?: Date;
  statut: 'actif' | 'inactif';
  createdAt: Date;
  updatedAt: Date;
}

/**
 * Statistiques d'un agent
 */
export interface AgentStats {
  id_agent: string;
  nom_complet: string;
  nb_operations: number;
  nb_operations_mois: number;
  poids_total: number;
  poids_mois: number;
  regions: string[];
  dernier_operation?: Date;
}

/**
 * Données de formulaire
 */
export type AgentFormData = Omit<Agent, 'id' | 'createdAt' | 'updatedAt' | 'regions' | 'operations'> & {
  regions: string[]; // IDs des régions
};

export type RegionFormData = Omit<Region, 'id' | 'createdAt' | 'updatedAt'>;

File: mobile_flutter/lib/models/agent.dart
class Agent {
  final int id;
  final String nom;
  final String prenom;
  final String? telephone;
  final String? email;
  final String username;
  final DateTime createdAt;
  final DateTime updatedAt;

  Agent({
    required this.id,
    required this.nom,
    required this.prenom,
    this.telephone,
    this.email,
    required this.username,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Agent.fromJson(Map<String, dynamic> json) {
    return Agent(
      id: json['id'],
      nom: json['nom'],
      prenom: json['prenom'],
      telephone: json['telephone'],
      email: json['email'],
      username: json['username'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nom': nom,
      'prenom': prenom,
      'telephone': telephone,
      'email': email,
      'username': username,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}
